bootstrap.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+# Bring up a server and install the Data Repository application either under
+# Vagrant or AWS.  For AWS install, the awscli software is expected to be
+# installed already on the local machine doing the bootstrapping, and the
+# appropriate AWS credentials set in the AWS_ACCESS_KEY and AWS_SECRET_KEY
+# environment variables.
+
+bootstrap_vagrant()
+{
+  # Run "varant up"
+  if [ -f Vagrantfile ]; then
+    echo "Running: vagrant up"
+    exec vagrant up
+  else
+    echo "No Vagrantfile in current directory to bootstrap via vagrant"
+    exit 1
+  fi
+}
+
+bootstrap_aws()
+{
+  # Create packed userdata bootstrap script
+  TMPDIR=`mktemp -d -t aws_bootstrap.XXXXX`
+  tar cJf ${TMPDIR}/files.tar.xz bootstrap_server.sh config.sh config_aws.sh install*.sh files
+  base64 ${TMPDIR}/files.tar.xz > ${TMPDIR}/files.b64
+  cat aws_bootstrapper_header.sh ${TMPDIR}/files.b64 > ${TMPDIR}/aws_bootstrapper.sh
+  # Bring up AWS instance
+  ID=$(aws ec2 run-instances --image-id $AWS_AMI \
+    --key-name $AWS_KEY_PAIR \
+    --security-group-ids $AWS_SECURITY_GROUP_IDS \
+    --subnet-id $AWS_SUBNET_ID \
+    --instance-type $AWS_INSTANCE_TYPE \
+    --associate-public-ip-address \
+    --user-data file://${TMPDIR}/aws_bootstrapper.sh \
+    --output text \
+    --query 'Instances[*].InstanceId')
+  # Wait for AWS server $ID to go out of "pending" state
+  /bin/echo -n "Waiting for server instance $ID to start: "
+  while state=$(aws ec2 describe-instances --instance-ids $ID --output text \
+      --query 'Reservations[*].Instances[*].State.Name'); test "$state" = "pending"; do
+    sleep 30
+    /bin/echo -n '.'
+  done; echo " $state"
+  # Associate elastic IP
+  aws ec2 associate-address --instance-id $ID --allocation-id $AWS_ELASTIC_IP
+  # Set descriptive name tag
+  aws ec2 create-tags --resources $ID --tags Key=Name,Value=$SERVER_HOSTNAME
+  # Clean up after ourselves
+  rm -rf $TMPDIR
+}
+
+# Process script arguments. Argument is server type: vagrant or aws; default: aws
+SCRIPTS_DIR="`pwd`"
+SERVER_ENV="vagrant"
```

Comment:
Good catch.  The comment is wrong.  Do you want me to submit a new pull request (re-typing in the humongous commit message :weary: ) or can I promise to fix it in the next update? :smiley: